app/controllers/posts_controller.rb

```diff
@@ -8,8 +8,64 @@ def index
   def show
     @user = User.find(params[:user_id])
     @user_post = @user.posts.find(params[:id])
+    @author_id = current_user.id
     @post_index = @user.posts.order(created_at: :desc).pluck(:id).index(@user_post.id) + 1
     @likes_count = @user_post.likes_counter
     @comments_count = @user_post.comments_counter
+    @comment = Comment.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+
+    if @post.save
+      flash[:success] = 'New post uploaded successfully!'
+      redirect_to user_post_url(current_user, @post)
+    else
+      flash[:error] = 'Post upload failed! Please try again.'
+      redirect_to new_user_post_url(current_user)
+    end
+  end
+
+  def create_comment
+    @user_post = Post.find(params[:id])
+    @comment = @user_post.comments.build(comment_params)
+    @comment.author = current_user
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @user_post.author_id, id: @user_post.id),
+                  notice: 'Comment created successfully.'
+    else
+      redirect_to user_post_path(user_id: @user_post.user.id, id: @user_post.id),
+                  alert: 'Failed to create comment.'
+    end
+  end
+
+  def like
+    @user_post = Post.find(params[:id])
+    @like = Like.new(author: current_user, post: @user_post)
+
+    if @like.save
```

Comment:
- [ ] Here also, for the like controller, we need to create a separate controller file `Like_controller` and put the create/like action inside this controller. Remember, a user should be able to add a like to posts. The likes controller should have a create actions.